tools/integration_tests/util/operations/file_operations.go

```diff
@@ -642,3 +632,60 @@ func SizeOfFile(filepath string) (size int64, err error) {
 
 	return (*fstat).Size(), nil
 }
+
+// Creates a temporary file (name-collision-safe) in /tmp with given content.
+// If gzipCompress is true, output file is a gzip-compressed file.
+// Caller is responsible for deleting the created file when done using it.
+// Failure cases:
+// 1. os.CreateTemp() returned error or nil handle
+// 2. gzip.NewWriter() returned nil handle
+// 3. Failed to write the content to the created temp file
+func CreateLocalTempFile(content string, gzipCompress bool) (string, error) {
+	contentSize := len(content)
+
+	// create appropriate name template for temp file
+	filenameTemplate := "testfile-*.txt"
+	if gzipCompress {
+		filenameTemplate += ".gz"
+	}
+
+	// create a temp file
+	f, err := os.CreateTemp(TmpDirectory, filenameTemplate)
+	if err != nil {
+		return "", fmt.Errorf("failed to create tempfile for template %s: %w", filenameTemplate, err)
+	} else if f == nil {
+		return "", fmt.Errorf("nil file handle returned from os.CreateTemp")
+	}
+	defer CloseFile(f)
+	filepath := f.Name()
+
+	if gzipCompress {
+		w := gzip.NewWriter(f)
+		if w == nil {
+			return "", fmt.Errorf("failed to create gzip writer for file %s", filepath)
+		}
+		defer func() {
+			if err := w.Close(); err != nil {
```

Comment:
It absorbs an existing non-nil value of err.

Reply:
here a new variable err is created in scope of if statement. hence it won't absorb existing value of err.